### PR TITLE
Engine Impl

### PR DESCRIPTION
**Pull request by junjiejiangjjj**
Engine impl
Combined test of engine/scheduler/task-executor/task  

#58

### CODE REVIEW

File: towhee/engine/task_scheduler.py [hunk 1]
```diff
@@ -40,30 +40,34 @@ class _TaskScheduler(ABC):
...
 
-        Args:
-            pipeline: `towhee.Pipeline`
-                A single pipeline to schedule tasks for.
-        """
-        pipeline.add_task_ready_handler(self.on_task_ready)
-        pipeline.add_task_start_handler(self.on_task_start)
-        pipeline.add_task_finish_handler(self.on_task_finish)
-        self._pipelines.append(pipeline)
+    # def add_pipeline(self, pipeline: Pipeline):
```
Comment (fzliu) [inline towhee/engine/task_scheduler.py:52]: Adding new pipelines after the `Engine` is created seems like a fairly important feature. Do we plan on adding this back later?

Comment (junjiejiangjjj) [reply towhee/engine/task_scheduler.py:52]: task_scheduler's  `self._pipelines` is the ref of pipelines, we only need call engine's `add_pipeline` function.

Comment (fzliu) [reply towhee/engine/task_scheduler.py:52]: Correct, but we also need to add the `Pipeline`'s task handlers, which we won't be able to do without wrapping it in a function.

Comment (junjiejiangjjj) [reply towhee/engine/task_scheduler.py:52]: In v0.1， we only have `write_output` handler to get one task's outputs. After v0.1, let us consider how to deal with other handlers.

**Pull request merged**